### PR TITLE
Multiple Issues and features

### PR DESCRIPTION
**Pull request by soycapitan**
Changes include:
- major architecture changes to feature multi complex actions, required to solve #21 
- hardcoded bind energy (a technical debt introduced to be resolved later), needed for future resolution of #22 
- removed sign correction feature, solves #23 
- changed legend in histograms not to use scientific notation, solves #20 
- created plotting feature as requested in #21 
- added information about number of datapoints used in 2D histograms, to address https://github.com/iitis/polymer_entropy/issues/17#issuecomment-1103584232

To craft plots run:
```
$python3 src/plot.py --datafolder data3 --complex 'Albumin+CS6'
```
All plots are available on [gdrive](https://drive.google.com/drive/folders/1Jx0cHh-51mVjVonenZNEjbG6O6vserbi?usp=sharing), samples below.

Albumin+CS6_4_analysis_Ca_hist2D_ϕ₁₄_ψ₁₄
plot21_Albumin+CS6

### CODE REVIEW

**Comment by soycapitan** [comment]
I have pushed more changes to fix some of our broken angles definitions, mainly to point that 14 angles are within single mer. Also changed column manipulation that proved to be broken because of my poor understanding of some python internals :(

Albumin+CS6_8_analysis_Na_hist2D_ϕ₁₄_ψ₁₄
Now histograms that should contain 24k data points do so.

Updated plots are available at gdrive in [PR24 r2](https://drive.google.com/drive/folders/1tIijxO97sHKC200ARBKmbcViwFHy7WEs?usp=sharing) directory.

The same command was used for plotting:
```
$python3 src/plot.py --datafolder data3 --complex 'Albumin+CS6'
```

To double-check that we are calculating the right things I have created `list_columns(self):` method and raised #25

File: src/experiment.py [hunk 1]
```diff
@@ -85,224 +119,248 @@ def drop_first_observations(self):
...
         plt.xlabel(angle_x)
         plt.ylabel(angle_y)
-        plt.title(f"Subsequent mers angles, R{self.num_realisation}")
-        if plotname:
-            plot_filepath = f"{plotname}hist2D_{self.chain.replace(' ','')}_{angle_x}_{angle_y}_b{numbins}.png"
-            plt.savefig(plot_filepath, dpi=self.plot_dpi)
-            plt.clf()
-        else:
-            plt.show()
+        plt.title(f"{self} - Subsequent mers angles, n={n_datapoints}")
```
Comment (kdomino) [inline src/experiment.py:176]: we can filter out analysis in f"{self}. We an also remove: "Subsequent mers angles" (I know I suggested it at the first place). The best  will be something like:   Albumin+CS6_realization:1_ion:Mg  or similar

**Review by kdomino — CHANGES_REQUESTED** [review]

File: src/experiment.py [hunk 2]
```diff
@@ -28,49 +30,81 @@ class Experiment:
...
+        ]
+        self.bind_energy = self.bind_energies[(self.complex, self.chain)][int(self.num_realisation)]
 
         if self.complex == 'Albumin+HA':
-            if self.chain == 'main chain':
+            if self.chain == 'analysis':
                 for mer in range(self.no_mers):
                     for angle in self.angles:
-                        self.columns.append(ColumnMeaning(f"{angle} mers {mer+1}, {mer+2}", "deg"))
+                        if angle.endswith('₁₃'):
```
[the remark this answers is not in the document]
Comment (soycapitan) [reply src/experiment.py:78]: Yes, it appears multiple times and I am thinking about using Python generators for this to keep it cleaner.

We need to _somehow_ (and this is probably not the best way) those angles as angles `13` are measured **between** consecutive mers and angles `14` are **inside** one mer, hence the difference implied in meaning and number of such angles (n-1 angles for angles between mers)/

[the remark this answers is not in the document]
Comment (soycapitan) [reply src/experiment.py:78]: I propose to create an issue out of recurring `if angle.endswith('₁₃'):` test not to impede ingesting working code.

[the remark this answers is not in the document]
Comment (kdomino) [reply src/experiment.py:78]: Ok I see. Perhaps for now making the function ```is_angle_between_mers``` or something

**Comment by soycapitan** [comment]
@kdomino ready for next review round

**Pull request merged**